src/panels/lovelace/entity-rows/hui-timer-entity-row.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  html,
+  LitElement,
+  TemplateResult,
+  property,
+  PropertyValues,
+} from "lit-element";
+
+import "../components/hui-generic-entity-row";
+
+import timerTimeRemaining from "../../../common/entity/timer_time_remaining";
+import secondsToDuration from "../../../common/datetime/seconds_to_duration";
+import { HomeAssistant } from "../../../types";
+import { EntityConfig } from "./types";
+import { HassEntity } from "home-assistant-js-websocket";
+
+class HuiTimerEntityRow extends LitElement {
+  @property() public hass?: HomeAssistant;
+  @property() private _config?: EntityConfig;
+  @property() private _timeRemaining?: number;
+  private _interval?: number;
+
+  public setConfig(config: EntityConfig): void {
+    if (!config) {
+      throw new Error("Configuration error");
+    }
+    this._config = config;
+  }
+
+  public disconnectedCallback(): void {
+    super.disconnectedCallback();
+    this._clearInterval();
+  }
+
+  protected render(): TemplateResult | void {
+    if (!this._config || !this.hass) {
+      return html``;
+    }
+
+    const stateObj = this.hass.states[this._config.entity];
+
+    if (!stateObj) {
+      return html`
+        <hui-error-entity-row
+          .entity="${this._config.entity}"
+        ></hui-error-entity-row>
+      `;
+    }
+
+    return html`
+      <hui-generic-entity-row .hass="${this.hass}" .config="${this._config}">
+        <div>${this._computeDisplay(stateObj)}</div>
+      </hui-generic-entity-row>
+    `;
+  }
+
+  protected updated(changedProps: PropertyValues) {
+    super.updated(changedProps);
+
+    if (changedProps.has("hass")) {
+      const stateObj = this.hass!.states[this._config!.entity];
+      const oldHass = changedProps.get("hass") as this["hass"];
+      const oldStateObj = oldHass ? oldHass.states[this._config!.entity] : "";
```

Comment:
`""` ? you mean undefined I think.

Reply:
@iantrich you forgot to address this comment

Reply:
My bad. Was rushing out the door. Will get to it in a few hours